FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +43,53 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] httpClientResult in
+
+			guard self != nil else { return }
+
+			var result: FeedLoader.Result
+
+			let feedLoaderResult = Self.httpClientResult2RemoteFeedLoaderResult(httpClientResult)
+			switch feedLoaderResult {
+			case .success(let responseData):
+				result = Self.responseData2FeedLoaderResult(responseData)
+
+			case .failure(let error):
+				result = .failure(error)
+			}
+
+			completion(result)
+		}
+	}
+
+	private static func httpClientResult2RemoteFeedLoaderResult(_ httpClientResult: HTTPClient.Result) -> RemoteFeedLoaderResult {
+		let feedLoaderResult = httpClientResult.mapError { _ in Error.connectivity }
+			.flatMap { (responseData, httpResponse) -> RemoteFeedLoaderResult in
+				guard httpResponse.isStatusOK else {
+					return .failure(Error.invalidData)
+				}
+				return .success(responseData)
+			}
+		return feedLoaderResult
+	}
+
+	private static func responseData2FeedLoaderResult(_ responseData: Data) -> FeedLoader.Result {
```

Comment:
Following from my previous comment, suggested name:
```suggestion
	private static func feedLoaderResult(from responseData: Data) -> FeedLoader.Result {
```